Dockerfile

```diff
@@ -1,30 +1,78 @@
 FROM tailor/java-oracle:latest
 MAINTAINER Nadav Shatz <nadav@tailorbrands.com>
 
-RUN apt-get update
-RUN apt-get install -y wget
-RUN apt-get install -y libc6-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*
+RUN add-apt-repository -y ppa:webupd8team/java
 
-# jruby
-ENV JRUBY_VERSION 9.1.8.0
-ENV JRUBY_SHA256 20ac501c99a7cb3cf53ded64ac1b8bb6e0b0f6ba34a41b8bacc9715cd4bb2601
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ trusty-pgdg main" > /etc/apt/sources.list.d/pgdg.list && \
+    wget --quiet -O - postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+
+RUN apt-get update && apt-get install -y git-core wget build-essential zlib1g-dev      \
+                       libssl-dev libreadline6-dev libyaml-dev libgdbm-dev libffi-dev  \
+                       libxml2-dev libxslt1-dev libncurses5-dev libfuse2 tcl-dev       \
+                       libpng-dev libjpeg-dev libgs-dev libbz2-dev libfreetype6-dev    \
+                       libtiff-dev librsvg2-dev libpq-dev postgresql-client-9.5        \
+                       curl apt-transport-https  libc6-dev --no-install-recommends
+
+RUN rm -rf /var/lib/apt/lists/*
+
+ENV JRUBY_VERSION 9.1.13.0
+ENV JRUBY_SHA256 9d156646623ac2f27174721035b52572a4b05690db7c1293295aa2c04aad3908
 RUN mkdir /opt/jruby \
-  && wget https://s3.amazonaws.com/jruby.org/downloads/${JRUBY_VERSION}/jruby-bin-${JRUBY_VERSION}.tar.gz -O /tmp/jruby.tar.gz \
+  && curl -fSL https://s3.amazonaws.com/jruby.org/downloads/${JRUBY_VERSION}/jruby-bin-${JRUBY_VERSION}.tar.gz -o /tmp/jruby.tar.gz \
   && echo "$JRUBY_SHA256 /tmp/jruby.tar.gz" | sha256sum -c - \
   && tar -zx --strip-components=1 -f /tmp/jruby.tar.gz -C /opt/jruby \
   && rm /tmp/jruby.tar.gz \
   && update-alternatives --install /usr/local/bin/ruby ruby /opt/jruby/bin/jruby 1
 ENV PATH /opt/jruby/bin:$PATH
 
-RUN echo 'gem: --no-rdoc --no-ri' >> /.gemrc
+# skip installing gem documentation
+RUN mkdir -p /opt/jruby/etc \
+    && { \
+        echo 'install: --no-document'; \
+        echo 'update: --no-document'; \
+    } >> /opt/jruby/etc/gemrc
+
+RUN gem install bundler
 
+# install things globally, for great justice
+# and don't create ".bundle" in all our apps
 ENV GEM_HOME /usr/local/bundle
-ENV PATH $GEM_HOME/bin:$PATH
-RUN gem install bundler \
-	&& bundle config --global path "$GEM_HOME" \
-	&& bundle config --global bin "$GEM_HOME/bin"
+ENV BUNDLE_PATH="$GEM_HOME" \
+    BUNDLE_BIN="$GEM_HOME/bin" \
+    BUNDLE_SILENCE_ROOT_WARNING=1 \
+    BUNDLE_APP_CONFIG="$GEM_HOME"
+ENV PATH $BUNDLE_BIN:$PATH
+RUN mkdir -p "$GEM_HOME" "$BUNDLE_BIN" \
+    && chmod 777 "$GEM_HOME" "$BUNDLE_BIN"
```

Comment:
quite aggressive setting this as 777 isn't it?

Reply:
Does it matter running inside the container? almost everything runs as root anyway no?